composer.json

```diff
@@ -18,8 +20,11 @@
     "extra": {
         "kimai": {
             "require": "1.9",
-            "version": "1.0.0",
+            "version": "1.0.1",
             "name": "ShareProjectTimesheetsBundle"
         }
+    },
+    "require-dev": {
+        "kevinpapst/kimai2": "^1.10"
```

Comment:
With this line my setup breaks. When executing `composer update` or `bin/console cache:clear` i receive:
```
PHP Fatal error:  Cannot declare class DoctrineExtensions\Query\MysqlWalker, because the name is already in use in <...>\kimai2\var\plugins\SharedProjectTimesheetsBundle\vendor\beberlei\doctrineextensions\src\Query\MysqlWalker.php on line 44
```
It's telling me that the plugins composer file required a php class that already exists - which is true, Kimai requires this class as well. What am i doing wrong? Any ideas @kevinpapst?

Reply:
You can remove the lines if you don't plan to use it in some CI server (eg GitHub actions like [here](https://github.com/Keleo/DemoBundle/blob/master/.github/workflows/linting.yaml)).

If you develop the bundle directly inside var/plugins/ then you don't ever need to execute composer and can ignore the require-dev.

I create every bundle in a separate directory and only deploy it to var/plugins/ to simulate the normal flow of every end-user.
And in that case the composer requirement is needed, so my IDE knows all sources.

Reply:
Thanks for your quick reply - got it, i'll set up the local deployment to follow the normal flow.